From f28afbc9ecf4b7cc72c708dda751202fc3f0eaf2 Mon Sep 17 00:00:00 2001
From: Jan Holthuis <jan.holthuis@ruhr-uni-bochum.de>
Date: Wed, 17 Aug 2022 23:29:24 +0200
Subject: [PATCH] Fix `-Wimplicit-function-declaration` errors when
 `DEBUG_KEXDH` is defined

Fixes the following errors:

    kexdh.c: In function 'kex_dh_compute_key':
    kexdh.c:83:2: warning: implicit declaration of function 'debug' [-Wimplicit-function-declaration]
       83 |  debug("bits %d", BN_num_bits(dh_pub));
          |  ^~~~~
    sshd.c: In function 'do_ssh2_kex':
    sshd.c:2293:2: warning: implicit declaration of function 'packet_start' [-Wimplicit-function-declaration]
     2293 |  packet_start(SSH2_MSG_IGNORE);
          |  ^~~~~~~~~~~~
    sshd.c:2294:2: warning: implicit declaration of function 'packet_put_cstring'; did you mean 'sshpkt_put_cstring'? [-Wimplicit-function-declaration]
     2294 |  packet_put_cstring("markus");
          |  ^~~~~~~~~~~~~~~~~~
          |  sshpkt_put_cstring
    sshd.c:2295:2: warning: implicit declaration of function 'packet_send' [-Wimplicit-function-declaration]
     2295 |  packet_send();
          |  ^~~~~~~~~~~
    sshd.c:2296:2: warning: implicit declaration of function 'packet_write_wait'; did you mean 'ssh_packet_write_wait'? [-Wimplicit-function-declaration]
     2296 |  packet_write_wait();
          |  ^~~~~~~~~~~~~~~~~
          |  ssh_packet_write_wait
---
 kexdh.c | 3 +++
 sshd.c  | 9 +++++----
 2 files changed, 8 insertions(+), 4 deletions(-)

diff --git a/kexdh.c b/kexdh.c
index 67133e33..531b8b60 100644
--- a/kexdh.c
+++ b/kexdh.c
@@ -42,6 +42,9 @@
 #include "digest.h"
 #include "ssherr.h"
 #include "dh.h"
+#ifdef DEBUG_KEXDH
+#include "log.h"
+#endif

 int
 kex_dh_keygen(struct kex *kex)
diff --git a/sshd.c b/sshd.c
index cbd3bce9..74011267 100644
--- a/sshd.c
+++ b/sshd.c
@@ -2290,10 +2290,6 @@ do_ssh2_kex(struct ssh *ssh)

 #ifdef DEBUG_KEXDH
 	/* send 1st encrypted/maced/compressed message */
-	packet_start(SSH2_MSG_IGNORE);
-	packet_put_cstring("markus");
-	packet_send();
-	packet_write_wait();
 #endif
 	debug("KEX done");
 }
--
2.37.1
